### PR TITLE
Updated Proto Files | Fixed FunctionsRequest | Added UserMan

### PR DESCRIPTION
**Pull request by AliabbasMerchant**

### CODE REVIEW

File: api/mgo/mongo.go [hunk 1]
```diff
@@ -126,3 +128,33 @@ func GenerateFind(condition utils.M) utils.M {
...
+// SignUp fires a signUp request
+func (s *Mongo) SignUp(email, name, password, role string) (*model.Response, error) {
+	m := &proto.Meta{DbType: s.db, Project: s.config.Project, Token: s.config.Token}
+	return s.config.Transport.SignUp(context.TODO(), m, email, name, password, role)
+}
+
+// EditProfile fires a editProfile request
+func (s *Mongo) EditProfile(id, email, name, password string) (*model.Response, error) {
+	m := &proto.Meta{DbType: s.db, Project: s.config.Project, Token: s.config.Token}
+	return s.config.Transport.SignUp(context.TODO(), m, id, email, name, password)
```
Comment (YourTechBud) [inline api/mgo/mongo.go]: Shouldn't this be editprofile

File: api/sql/sql.go [hunk 2]
```diff
@@ -42,3 +44,33 @@ func (s *SQL) Update(col string) *Update {
...
+// SignUp fires a signUp request
+func (s *SQL) SignUp(email, name, password, role string) (*model.Response, error) {
+	m := &proto.Meta{DbType: s.db, Project: s.config.Project, Token: s.config.Token}
+	return s.config.Transport.SignUp(context.TODO(), m, email, name, password, role)
+}
+
+// EditProfile fires a editProfile request
+func (s *SQL) EditProfile(id, email, name, password string) (*model.Response, error) {
+	m := &proto.Meta{DbType: s.db, Project: s.config.Project, Token: s.config.Token}
+	return s.config.Transport.SignUp(context.TODO(), m, id, email, name, password)
```
Comment (YourTechBud) [inline api/sql/sql.go]: Shouldn't this be Editprofile

File: api/mgo/mongo.go [hunk 3]
```diff
@@ -126,3 +128,33 @@ func GenerateFind(condition utils.M) utils.M {
...
+}
+
+// SignUp fires a signUp request
+func (s *Mongo) SignUp(email, name, password, role string) (*model.Response, error) {
+	m := &proto.Meta{DbType: s.db, Project: s.config.Project, Token: s.config.Token}
+	return s.config.Transport.SignUp(context.TODO(), m, email, name, password, role)
+}
+
+// EditProfile fires a editProfile request
+func (s *Mongo) EditProfile(id, email, name, password string) (*model.Response, error) {
```
Comment (YourTechBud) [inline api/mgo/mongo.go]: Shouldn't we take email, name and password as a struct. Passing it as parameters makes them look compulsory. A struct on the other hand can have fields left unset. Lets make this change for SQL as well. We should do this in python too.

Comment (YourTechBud) [reply api/mgo/mongo.go]: Im not sure about this though. @AliabbasMerchant what do you think?

**Review by YourTechBud — CHANGES_REQUESTED** [review]
Minor changes.

File: api/mgo/mongo.go (hunk 1 above)
Comment (AliabbasMerchant) [reply api/mgo/mongo.go]: Oops 🙈

File: api/mgo/mongo.go (hunk 3 above)
Comment (AliabbasMerchant) [reply api/mgo/mongo.go]: Even I was unsure about this.  
(In python I have kept them as optional, so no prob)  
I think a struct should be ok.

```
type NewValues struct {
    Email string
    Name string
    Password string
}
```

**Comment by AliabbasMerchant** [comment]
@YourTechBud Done. Please have a look.

**Comment by YourTechBud** [comment]
Can we never the last 2 commits into one?

**Comment by AliabbasMerchant** [comment]
@YourTechBud  Done

**Pull request merged**